src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/NotYetReady/CountA.txt

```diff
@@ -0,0 +1,338 @@
+2
+
+>> CountA(Table({a:"1"},{a:"2"},If(1<0,{a:"3"}),If(1>0,{a:"4"}),{a:"5"}))
+4
+
+>> CountA(LastN([1,2,3,4], Blank()))
+Blank()
+
+>> CountA([LastN([1,2,3,4]), Blank()])
+1
+
+//One blank record
+>> CountA([Blank()])
+0
+
+//All blank records
+﻿>> CountA([Blank(),Blank(),Blank(),Blank(),Blank()])
+0
+
+>> CountA(Table({a:Blank()},{a:Blank()},{a:Blank()}))
+0
+
+﻿>> CountA(["",""," ",Blank(),Blank()])
+3
+
+>> CountA(Table({a:""},{a:" "},{a:Blank()}))
+2
+
+// ******** NULL and ERROR RECORDS ********
+
+//Few error records
+>> CountA([1, "2", 1/0, true, false, Blank(), "", " ", Date(2022,12,12),Time(6,30,30),DateTimeValue("5/12/2022 6:30:30 PM")])
+#Error(Kind=Div0)
+
+>> CountA(Table({a:1},{a:2},If(Sqrt(-1)<1,{a:3}),{a:4}))
+#Error(Kind=Numeric)
+
+//All error records
+>> CountA([1/0])
+#Error(Kind=Div0)
+
+>> CountA(Table({a:1/0},{a:Sqrt(-1)}))
+#Error(Kind=Div0)
+
+>> CountA(Sort(["-2","-1","0",1,2],1/Value))
+#Error(Kind=Div0)
+
+>> CountA([Date(2022,12,12), Time(6,30,30), 1/0, DateTimeValue("5/12/2022 6:30:30 PM")])
+#Error(Kind=Div0)
+
+// ************************************** COERCION CASES **************************************
+
+// Note: CountA function accepts different dataype only in the array notation. In the table notation, all the records should be of the same datatype. Thus coercion cases will be applicable only for the array notation expressions 
+
+// ====================== FIRST RECORD NUMBER, SECOND RECORD OTHER DATAYPE ======================
+
+//Number-Number
+>> CountA([1,2])
+2
+
+>> CountA(Table({a:1},{a:2}))
+2
+
+//Number-String
+>> CountA([1,"2"])
+2
+
+>> CountA([1,"two"])
+#Error(Kind=InvalidArgument)
+
+//Number-Boolean
+>> CountA([1,true])
+2
+
+>> CountA([1,false])
+2
+
+//Number-Date
+>> CountA([1,Date(2022,12,12)])
+2
+
+>> CountA([1,DateValue("May 12, 2022")])
+2
+
+
+//Number-Time
+>> CountA([1,Time(6,30,30)])
+2
+
+>> CountA([1,TimeValue("12:15 PM")])
+2
+
+//Number-DateTimeValue
+>> CountA([1,DateTimeValue("5/12/2022 6:30:30 PM")])
+2
+
+// ====================== FIRST RECORD STRING, SECOND RESULT OTHER DATAYPE ======================
+
+//String-Number
+>> CountA(["Hello",2])
+2
+
+//String-String
+>> CountA(["Hello","2"])
+2
+
+>> CountA(Table({a:"Hello"},{a:"2"}))
+2
+
+>> CountA(["Hello","two"])
+#Error(Kind=InvalidArgument)
+
+//String-Boolean
+>> CountA(["Hello",true])
+2
+
+>> CountA(["Hello",false])
+2
+
+//String-Date
+>> CountA(["Hello",Date(2022,12,12)])
+2
+
+>> CountA(["Hello",DateValue("May 12, 2022")])
+2
+
+
+//String-Time
+>> CountA(["Hello",Time(6,30,30)])
+2
+
+>> CountA(["Hello",TimeValue("12:15 PM")])
+2
+
+//String-DateTimeValue
+>> CountA(["Hello",DateTimeValue("5/12/2022 6:30:30 PM")])
+2
+
+// ====================== FIRST RECORD BOOLEAN, SECOND RESULT OTHER DATAYPE ======================
+
+//Boolean-Number
+>> CountA([true,2])
+2
+
+//Boolean-String
+>> CountA([true,"2"])
+2
+
+>> CountA([true,"two"])
+#Error(Kind=InvalidArgument)

```

Comment:
This should not be invalid - "two" would be coerced to a boolean value (false), and the result should be 2

Reply:
Corrected the expected result and updated to 2